auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -185,9 +234,25 @@ public void onFailure(Bundle extra) {
     @Override
     public void onClick(View view) {
         if (view.getId() == R.id.email_provider) {
-            startActivityForResult(
-                    RegisterEmailActivity.createIntent(this, mActivityHelper.getFlowParams()),
-                    RC_EMAIL_FLOW);
+            if (mActivityHelper.getFlowParams().isReauth) {
+                String reauthEmail = mActivityHelper.getFlowParams().reauthEmail;
+                if (reauthEmail == null) {
+                    Log.e(TAG, "Attempting to reauth with email but no reauthEmail provided");
+                    return;
+                }
+                startActivityForResult(
+                        WelcomeBackPasswordPrompt.createIntent(
+                                this,
+                                mActivityHelper.getFlowParams(),
+                                new IdpResponse(
+                                        AuthUI.EMAIL_PROVIDER,
+                                        reauthEmail)),
+                        RC_EMAIL_DIRECT_SIGN_IN);
+            } else {
+                startActivityForResult(
+                        RegisterEmailActivity.createIntent(this, mActivityHelper.getFlowParams()),
```

Comment:
@amandle I'm just being nit picky here (Sorry! 😄), but I feel like `mActivityHelper.getFlowParams()` could be `Ctrl + Alt + V`ed into something like `params`.